lenskit-core/src/main/java/org/lenskit/basic/TopNItemBasedItemRecommender.java

```diff
@@ -89,21 +84,15 @@ protected LongSet getDefaultExcludes(LongSet items) {
      * @return The top <var>n</var> items from <var>scores</var>, in descending
      *         order of score.
      */
-    protected List<ScoredId> recommend(int n, SparseVector scores) {
+    protected ResultList recommend(int n, ResultMap scores) {
         if (scores.isEmpty()) {
-            return Collections.emptyList();
+            Results.newResultList();
         }
 
         if (n < 0) {
             n = scores.size();
         }
 
-        ScoredItemAccumulator accum = new TopNScoredItemAccumulator(n);
-        for (VectorEntry pred : scores) {
-            final double v = pred.getValue();
-            accum.put(pred.getKey(), v);
-        }
-
-        return accum.finish();
+        return Results.newResultList(Results.scoreOrder().greatestOf(scores, n));
```

Comment:
In my mind ordering concerns and limit concerns should be two parameters. That said, if this is guava stuff ignore my concerns.